modules/cloudfront_distribution/tf_module/main.tf

```diff
@@ -7,9 +8,68 @@ data "aws_s3_bucket" "logging_bucket" {
   bucket = var.s3_log_bucket_name
 }
 
+locals {
+  lb_distribution_count = var.eks_load_balancer_enabled == true ? 1 : 0
+  s3_distribution_count = var.s3_load_balancer_enabled == true ? 1 : 0
+}
+
+resource "aws_cloudfront_distribution" "lb_distribution" {
+  count = local.lb_distribution_count
+  origin {
+    domain_name = var.load_balancer
+    origin_id   = local.lb_origin_id
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "http-only"
+      origin_ssl_protocols   = ["TLSv1.2", "TLSv1.1", "TLSv1"]
+    }
+  }
+
+  comment         = "Opta managed cloudfront distribution ${var.layer_name}-${var.module_name}"
+  enabled         = true
+  is_ipv6_enabled = true
+
+  dynamic "logging_config" {
+    for_each = var.s3_log_bucket_name == null ? [] : [1]
+    content {
+      include_cookies = true
+      bucket          = data.aws_s3_bucket.logging_bucket[0].bucket_domain_name
+      prefix          = "cloudfront/${var.layer_name}/${var.module_name}"
+    }
+  }
+
+  default_cache_behavior {
+    allowed_methods        = ["GET", "HEAD", "OPTIONS"]
```

Comment:
what's the difference b/w allowed and cached? and yes, it's good idea to expose the cached methods to users but as optional. and you can use this list as default if the user doesn't specify it

Reply:
`allowed_methods`: Controls which HTTP methods CloudFront processes and forwards to your Amazon S3 bucket or your custom origin.
`cached_methods`: Controls whether CloudFront caches the response to requests using the specified HTTP methods.

Can give access to the users for both of them. Or should we do that in the next phase?